### ISSUE TITLE
Add ICL ig-platform-id

### ISSUE THREAD

**Issue by 1alessandro1** [acidanthera/bugtracker#832]
Hi, good job adding the new MacBookAir9,1 SMBIOSes to support ice lake. 
Can you please post here the applelife.ru topic which collected the darwindumper results on all the machines or add ig-platform-id and device-id for G5 and G7 intel iris plus Graphics? 

Thanks.

**Comment by vit9696** [comment]
I am not sure we have it accessible for public. cc @Andrey1970AppleLife 

Andrey, we should still update our manuals on AppleLife and WEG Docs. Also, I would like us to have the default connector-less configurations to be obvious in the manuals. I remember some connector-less framebuffers working worse than the other.

Also, now that ICL and others got released, could you please review our
- fallback framebuffers for connectors and without them are still correct
- the list of connector-less framebuffers is accurate (we made them in early beta after all)

REF:
https://github.com/acidanthera/Lilu/blob/3b51df3/Lilu/Headers/kern_devinfo.hpp#L168-L200
https://github.com/acidanthera/Lilu/blob/3b51df3/Lilu/Sources/kern_devinfo.cpp#L46-L111

**Comment by Andrey1970AppleLife** [comment]
Done

**Comment by 1alessandro1** [comment]
Good evening, thanks for providing the new framebuffers 

I have a little doubt on this matter...
Is it normal that no framebuffers have fbmem different from zero? (all are marked as "nofbmem")
Also, why in the recommended framebuffers there's only this recommendation?

"Recommended framebuffers: for laptop - 0x8A520000 (default). "

While on different iGPUs you can find the device-ids which are recommended as well..

Thanks in advance for clarifying.

**Comment by Andrey1970AppleLife** [comment]
no fbmem it is normal, look at other generations, there too such, for example CFL. 
TOTAL STOLEN 193 MB "?" 
In BIOS it is necessary equally or more. But this value while in doubt. "?"

0x8A520000 it by simply default. Don't read it recommended, we have no confirmations for this.

Fake device-id it is necessary only in special cases which are specified in IntelFAQ.

**Comment by 1alessandro1** [comment]
Currently I have a Lenovo S740 14IIL with 

CPU: i7 1065G7 (The MacbookAir9,1 has an i7 1060NG7 so i guess the iGPU is the same)
RAM: 16GB 3733MHz
M.2: the PM981a 256GB (I think I'll replace it to reach 100% compatibility)

I've frequently read from other lazy issues opened inside bugtracker that I shoud use forums for support, this post describes what I experienced so far with the guys on macos86.it. 

[https://www.insanelymac.com/forum/topic/331381-aptiomemoryfix/?do=findComment&comment=2717108](https://www.insanelymac.com/forum/topic/331381-aptiomemoryfix/?do=findComment&comment=2717108)

Any help is appreciated, if we work together, can we at least manage to boot in -igfxvesa on this ICL laptop? Or maybe waiting for the next OpenCore 0.5.8, Lilu, WhateverGreen, AppleALC and VirtualSMC would help somehow?

Thanks

Ale

**Comment by vit9696** [comment]
@1alessandro1 nobody has ICL on our side at the moment, so I cannot say whether it is supported or not. Basically I would say that nothing above Z370 is supported and that you are on your own with such hardware for the nearest future. We have several other more important issues to nail first before we can start looking for new hardware.

**Comment by 1alessandro1** [comment]
@vit9696 
> nobody has ICL on our side at the moment
Sorry to disturb, but what do you actually require in order to be able to test ICL support (in all its different shapes and forms?

For example, are legacy apple hardware the main source of information (such as the MBA9,1) or do you need also normal ICL laptops like the one I'm currently using?
Are donations the only way to buy and to obtain those infos or can you get what you actually need from just a normal MacBook9,1 that (for example) a friend of mine has?

In other words, what else can be done in order to fulfill your needs and be able to debug/test what's required (couple more SSDTs as Z390 demonstrated with SSDT-PMC and other quirks - protectuefiservices, devirtualizemmio etc) and what is mandatory for you to get all the infos that allow you to say: "yeah, I can find a workaround for that - or - that compromises the whole installation process"?

Thanks

**Comment by vit9696** [comment]
@1alessandro1 we do not have needs in ICL, that's pretty much it. Are you really insane to think we are looking for more work for nothing? Pretty much nobody here cares about the hardware we do not have. Especially when it needs extra work that has little benefits. That's basically why stuff like Z390, AMD, or this ICL gets so little love.

**Comment by frozenzero123** [comment]
> 目前我有一个联想S740 14IIL
> 
> CPU：i7 1065G7（MacbookAir9,1拥有i7 1060NG7，所以我想iGPU是相同的）
> RAM：16GB 3733MHz
> M.2：PM981a 256GB（我想我将替换它以达到100％的兼容性）
> 
> 我经常阅读在bugtracker中打开的其他懒惰问题，我应该使用论坛来寻求支持，这篇文章描述了到目前为止我在macos86.it上所遇到的问题。
> 
> https://www.insanelymac.com/forum/topic/331381-aptiomemoryfix/?do=findComment&comment=2717108
> 
> 感谢您提供任何帮助，如果我们一起工作，至少可以在此ICL笔记本电脑上设法在-igfxvesa中启动吗？或者也许等待下一个OpenCore 0.5.8，Lilu，WhateverGreen，AppleALC和VirtualSMC会有所帮助吗？
> 
> 谢谢
> 
> 麦酒

You and I are the same model, and currently have the same problem. Although it can enter the system without driving the graphics card, as long as the correct device-id is input, it will panic and display cd clock.

**Comment by frozenzero123** [comment]
> Done

Hello, we are doing the installation of 10nmice lake, and we have made some progress, but the only thing that can't be helped is the graphics card. Now it is driven by Koreans (1035G4) but they have not made special settings, and we are now stuck in panic. I don't know how to solve it, I would like to ask you, I hope you can help us!
[s740 panic.txt](https://github.com/acidanthera/bugtracker/files/5066790/s740.panic.txt)

**Comment by frozenzero123** [comment]
> no fbmem it is normal, look at other generations, there too such, for example CFL.
> TOTAL STOLEN 193 MB "?"
> In BIOS it is necessary equally or more. But this value while in doubt. "?"
> 
> 0x8A520000 it by simply default. Don't read it recommended, we have no confirmations for this.
> 
> Fake device-id it is necessary only in special cases which are specified in IntelFAQ.
> What full name of yours CPU?
> 
> ICL IGPU driver strongly changed. I don't know anybody who ICL IGPU launched. Don't hope that will be simple.

Hello, we are doing the installation of 10nmice lake, and we have made some progress, but the only thing that can't be helped is the graphics card. Now it is driven by Koreans (1035G4) but they have not made special settings, and we are now stuck in panic. I don't know how to solve it, I would like to ask you, I hope you can help us!
[s740 panic.txt](https://github.com/acidanthera/bugtracker/files/5066796/s740.panic.txt)
